enclave/include/enclave_context.h

```diff
@@ -184,95 +219,87 @@ class EnclaveContext {
-    //    LOG(INFO) << "verification failed - Could not read key";
-    //    LOG(INFO) << "verification failed - mbedtls_pk_parse_public_keyfile returned" << ret;
-    //    return false;
-    //  }
-    //
-    //  if(!mbedtls_pk_can_do(&_pk_context, MBEDTLS_PK_RSA)) {
-    //    LOG(INFO) << "verification failed - Key is not an RSA key";
-    //    return false;
-    //  }
-    //
-    //  mbedtls_rsa_set_padding(mbedtls_pk_rsa(_pk_context), MBEDTLS_RSA_PKCS_V21, MBEDTLS_MD_SHA256);
-    //
-    //  if((ret = compute_sha256(data, data_len, hash)) != 0) {
-    //    LOG(INFO) << "verification failed -- Could not hash";
-    //    return false;
-    //  }
-    //
-    //  if((ret = mbedtls_pk_verify(&_pk_context, MBEDTLS_MD_SHA256, hash, 0, signature, sig_len)) != 0) {
-    //    LOG(INFO) << "verification failed -- mbedtls_pk_verify returned " << ret;
-    //    return false;
-    //  }
-    //
-    //  return true;
-    //}
-
-    void sync_client_key() {
-      // The master node (rank 0) broadcasts the client key to other nodes
-      rabit::Broadcast(client_key, CIPHER_KEY_SIZE, 0);
-      client_key_is_set = true;
+    char* get_client_cert(char *username) {
+      LOG(DEBUG) << "Getting username " << username;
+      std::string str(username);
+      auto iter = client_public_keys.find(str);
+      if (iter == client_public_keys.end()) {
+        LOG(FATAL) << "No certificate for user: " << username;
+      } else {
+        return (char*) iter->second.data();
+      }
     }
 
-    //bool decrypt_and_save_client_key(uint8_t* data, size_t data_len, uint8_t* signature, size_t sig_len) {
-    //  if (rabit::GetRank() == 0) {
-    //    if (!verifySignature(data, data_len, signature, sig_len)) {
-    //      LOG(INFO) << "Signature verification failed";
-    //      return false;
-    //    }
-    //
-    //    int res = 0;
-    //    mbedtls_rsa_context* rsa_context;
-    //
-    //    mbedtls_pk_rsa(m_pk_context)->len = data_len;
-    //    rsa_context = mbedtls_pk_rsa(m_pk_context);
-    //    rsa_context->padding = MBEDTLS_RSA_PKCS_V21;
-    //    rsa_context->hash_id = MBEDTLS_MD_SHA256;
-    //
-    //    size_t output_size;
-    //    uint8_t output[CIPHER_KEY_SIZE];
-    //
-    //    res = mbedtls_rsa_pkcs1_decrypt(
-    //        rsa_context,
-    //        mbedtls_ctr_drbg_random,
-    //        &m_ctr_drbg_context,
-    //        MBEDTLS_RSA_PRIVATE,
-    //        &output_size,
-    //        data,
-    //        output,
-    //        CIPHER_KEY_SIZE);
-    //    if (res != 0) {
-    //      LOG(INFO) << "mbedtls_rsa_pkcs1_decrypt failed with " << res;
-    //      return false;
-    //    }
-    //    std::vector<uint8_t> v(output, output + CIPHER_KEY_SIZE);
-    //    memcpy(client_key, output, CIPHER_KEY_SIZE);
-    //    client_key_is_set = true;
-    //  }
-    //  sync_client_key();
-    //  return true;
-    //}
+    /*
+     *void sync_client_key() {
```

Comment:
Let's remove this `sync_client_key()` function? If we don't use it here I don't think we need it -- I also don't use it in my multiclient cluster changes.